src/contracts/atlas/AtlasVerification.sol

```diff
@@ -161,6 +186,10 @@ contract AtlasVerification is EIP712, DAppIntegration {
                 return (userOpHash, ValidCallsResult.InvalidBundler);
             }
 
+            if (dAppOp.control != userOp.control) {
+                return (userOpHash, ValidCallsResult.InvalidControl);
+            }
```

Comment:
```suggestion
```
I don't think this check is needed/reachable either. We get the dConfig by reading it from the contract deployed at `userOp.control` and then, above this, on L156 we would catch that `dAppOp.control` is not the same address in `_verifyDApp` where it checks `dAppOp.control == dConfig.to` on L365